### PR TITLE
feat: add go bindings, build with release mode, inline cgo build flags

### PR DESCRIPTION
**Pull request by b5**
~~I took a stab at updating the go bindings after merging #20, came across a gnarly error running the updated main.go:~~ turns out I was just running a super outdated bindgen fork.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +72,8 @@ Install `uniffi-bindgen-go`:
 cargo install uniffi-bindgen-go --git https://github.com/dignifiedquire/uniffi-bindgen-go --branch upgarde-uniffi-24
 ```
 
+run `./make_go.sh` from the root directory. This will generate a file at `./go/go/iroh.go` with a large comment block that needs to be cut up into `irohFFI.h`. move the remainder of the code up into `iroh.go` & remove the `iroh/iroh` directory.
```
Comment (dignifiedquire) [inline README.md:75]: yeah these instructions are straight not true, no cutting up of comment blocks needed

File: src/node.rs [hunk 2]
```diff
@@ -251,14 +251,20 @@ impl Doc {
         author_id: Arc<AuthorId>,
         key: Vec<u8>,
         value: Vec<u8>,
-    ) -> Result<Arc<Hash>, Error> {
+    ) -> Result<Arc<Entry>, Error> {
         block_on(&self.rt, async {
-            let hash = self
-                .inner
-                .set_bytes(author_id.0.clone(), key, value)
+            // TODO - set_bytes should return an entry? probably.
```
Comment (dignifiedquire) [inline src/node.rs:256]: cc @Frando 

I think we should unifiy the API around either `Hash` or `Entry`. And we need a method to go from `Hash` to `Entry`, currently things are a little bit wonky here

**Comment by dignifiedquire** [comment]
https://github.com/n0-computer/iroh-ffi/pull/19 is now up to date

**Comment by b5** [comment]
closed in favour of #19

**Pull request closed without merging**